src/epcpm/staticmodbustoxls.py

```diff
@@ -105,26 +134,84 @@ class Root:
     def gen(self) -> openpyxl.workbook.workbook.Workbook:
         """
         Excel spreadsheet generator for the static modbus root class.
-
         Returns:
             workbook: generated Excel workbook
         """
         workbook = openpyxl.Workbook()
         workbook.remove(workbook.active)
-        worksheet = workbook.create_sheet()
-        worksheet.append(field_names.as_filtered_tuple(self.column_filter))
+        modbus_worksheet = workbook.create_sheet("Static Modbus Data")
+        enumeration_worksheet = workbook.create_sheet("Enumerations")
+        modbus_worksheet.append(field_names.as_filtered_tuple(self.column_filter))
+        enumeration_worksheet.append(["Enumerator", "Name", "Value"])
+
+        scale_factor_from_uuid = build_uuid_scale_factor_dict(
+            points=self.wrapped.children,
+            parameter_uuid_finder=self.parameter_uuid_finder
+        )
+        enumerations = self.collect_enumerations()
+        enumerators_from_uuid = {}
+        for enumeration in enumerations:
+            enumerators_from_uuid[enumeration.uuid] = enumeration
+
+        # Collects the cell coordinates of each new enumerator as they're appended
+        enumeration_cell_coordinates = {}
+        ENUMERATION_ENUMERATOR_COLUMN = "#Enumerations!A"
+        for enumeration in enumerations:
+            for child in enumeration.children:
+                enumeration_worksheet.append([enumeration.name, child.name, child.value])
+                if enumeration.name not in enumeration_cell_coordinates.keys():
+                    enumeration_cell_coordinates[enumeration.name] = (
+                        ENUMERATION_ENUMERATOR_COLUMN + f"{enumeration_worksheet.max_row}"
+                    )
 
         for member in self.wrapped.children:
             builder = builders.wrap(
                 wrapped=member,
                 parameter_uuid_finder=self.parameter_uuid_finder,
+                scale_factor_from_uuid=scale_factor_from_uuid,
+                enumerators_from_uuid=enumerators_from_uuid
             )
             rows = builder.gen()
+
+            # Appends the rows of static modbus data
+            MODBUS_ENUMERATOR_COLUMN = "O"
             for row in rows:
-                worksheet.append(row.as_filtered_tuple(self.column_filter))
+                modbus_worksheet.append(row.as_filtered_tuple(self.column_filter))
+                # Checks if there's an enumerator and hyperlinks it to the right row in
+                # the Enumerations worksheet
+                if row.enumerator is not None:
+                    max_row = modbus_worksheet.max_row
+                    if row.enumerator in enumeration_cell_coordinates.keys():
+                        modbus_worksheet[MODBUS_ENUMERATOR_COLUMN +f"{max_row}"].hyperlink = (
+                            enumeration_cell_coordinates[row.enumerator]
+                        )
 
         return workbook
 
+    def collect_enumerations(self):
```

Comment:
Can you add a return type hint and docstring for `collect_enumerations`, please?

Reply:
[5aea6b6](https://github.com/epcpower/pm/pull/71/commits/5aea6b6671bfebe84a18b153f896d3b0bfbebdf3)